src/python/pants/option/global_options.py

```diff
@@ -463,7 +472,15 @@ def register_bootstrap_options(cls, register):
             "your machine or when they are ignored by the `--pants-ignore` option.",
         )
 
-        # TODO(#7203): make a regexp option type!
+        register(
```

Comment:
Making this a bootstrap option means that the scheduler will need to re-initialize in order for it to change... an alternative might be to make the `AddressMapper` a root parameter via `RootRule`, or to inject the filtering options in a separate parameter.

Reply:
A followup PR stops special-casing `AddressMapper` being a singleton rule and makes it a normal rule defined in `build_files.py` that takes `GlobalOptions` as a parameter. I think that we can fix this in that followup.